datasaur-ai/robosaur
# #10838 Added support to use script from PCW

## Changes
* Added `pcwPayloadSource` and `pcwPayload` to Config interface
* Added `PCWTransformer` and its mapper files
* Updated `README.md`

## How to test
To use the script generated from PCW use the option `--use-pcw` on `create-projects` command.

```bash
npm run start -- create-projects <path-to-config-file> --use-pcw
```

When using `--use-pcw` option, provide both `pcwPayloadSource` and `pcwPayload` inside `project` in the config file.

For further instructions, see `README.md`

## Notes
Currently the only thing **not** supported is document assignment. **Need further discussion to implement this**.

Problem:
* In robosaur, document assignments requires team_member **email**, while in PCW, only `team_member_id` is provided

Proposed Solution:
* Create a new query to get the user's email from `team_member_id`
* Give exception in documentation to not use PCW's document assignment

## UPDATE 1

document assignment through PCW payload is now supported. For further information see `README.md`
